agents/broker.py

```
@@ -0,0 +1,105 @@
+import threading
+import uuid
+from contextlib import contextmanager
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+_LEDGER_LOCK = threading.Lock()
+
+
+@contextmanager
+def _ledger_file_lock(ledger_path: Path):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_ledger_file_lock`

(ANN202)